addons/portal/data/mail_template_data.xml

```diff
@@ -5,10 +5,14 @@
         <record id="mail_template_data_portal_welcome" model="mail.template">
             <field name="name">Portal: User Invite</field>
             <field name="model_id" ref="portal.model_portal_wizard_user"/>
-            <field name="subject">Your account at {{ object.user_id.company_id.name }}</field>
-            <field name="email_to">{{ object.user_id.email_formatted }}</field>
+            <field name="subject">Your account at {{ object.user_id.company_id.name or object.partner_id.company_id.name or ''}}</field>
+            <field name="email_to">{{ object.user_id.email_formatted or object.partner_id.email_formatted}}</field>
             <field name="description">Invitation email to contacts to create a user account</field>
             <field name="body_html" type="html">
+            <!-- defaulting to the partner as fallback for preview wizard (used when user hasn't been created yet) -->
+            <t t-set="user_name" t-value="object.user_id.name or object.partner_id.name or ''"/>
+            <t t-set="company_name" t-value="object.user_id.company_id.name or object.partner_id.company_id.name or ''"/>
+            <t t-set="signup_url" t-value="object.user_id.partner_id._get_signup_url() or object.partner_id._get_signup_url()"/>
```

Comment:
I think the original issue (traceback) is because `_get_signup_url` calls `ensure_one` so we'd probably need to check `user_id.partner_id` is set first

If I'm not mistaken

Reply:
@nd-dew :fish:

Reply:
@nd-dew 
Why not I guess while we're making it work for user-less wizard for everything else ^^'

Would add parentheses though